lib/recursive-methods.rb

```diff
@@ -1,49 +1,125 @@
 # Authoring recursive algorithms. Add comments including time and space complexity for each method.
-
-# Time complexity: ?
-# Space complexity: ?
-def factorial(n)
-    raise NotImplementedError, "Method not implemented"
+require "pry"
+# Time complexity: O(n)
+# Space complexity: O(n)
+def factorial(num)
+  raise ArgumentError if num < 0
+  if num == 1 || num == 0 # base case
+    return 1
+  else
+    return num * factorial(num - 1)
+  end
 end
 
-# Time complexity: ?
-# Space complexity: ?
+# O(n) for both T & S it's decrementing by 1, the function calls are the same for both
+
+# Time complexity: O(n)
+# Space complexity: O(n)
 def reverse(s)
```

Comment:
The line `s[1..-1]` creates a new array.  So this makes it O(n^2) in space and time complexity.